singer_sdk/helpers/_state.py

```diff
@@ -217,7 +218,7 @@ def increment_state(
         progress_dict = stream_or_partition_state[PROGRESS_MARKERS]
     old_rk_value = to_json_compatible(progress_dict.get("replication_key_value"))
     new_rk_value = to_json_compatible(latest_record[replication_key])
-    if old_rk_value is None or new_rk_value >= old_rk_value:
+    if old_rk_value is None or new_rk_value >= old_rk_value or not check_sorted:
```

Comment:
@edgarrmondragon - Just as a small matter, and probably a nit, what do you think of us putting the `or not check_sorted` ahead of the `>=` comparison. The difference would be small but this would prevent comparison errors in case that for some reason the values do not support comparison.

Reply:
@aaronsteers ah that makes sense, do start a PR. Or, I'll do it after lunch 🥪 .

Reply:
Tracked here: https://github.com/meltano/sdk/issues/736

Reply:
Thanks, @edgarrmondragon !